test/specs/elements/Label/Label-test.js

```diff
@@ -24,176 +25,106 @@ describe('Label Component', () => {
   common.propKeyAndValueToClassName(Label, 'attached')
 
   common.propKeyOrValueToClassName(Label, 'corner')
-  common.propKeyOrValueToClassName(Label, 'pointing')
   common.propKeyOrValueToClassName(Label, 'ribbon')
 
   common.implementsIconProp(Label)
+  common.implementsImageProp(Label)
 
   it('is a div by default', () => {
     shallow(<Label />)
       .should.have.tagName('div')
   })
 
-  describe('detail', () => {
-    it('renders detail as a div', () => {
-      shallow(<Label detail={faker.hacker.noun()} />)
-        .find('.detail')
-        .should.have.tagName('div')
-    })
-
-    it('renders detail using component from detailAs prop', () => {
-      shallow(<Label detail={faker.hacker.noun()} detailAs='a' />)
-        .find('.detail')
-        .should.have.tagName('a')
-    })
-
-    it('has no detail when not defined', () => {
+  describe('content', () => {
+    it('has no content by default', () => {
       shallow(<Label />)
-        .should.not.have.descendants('.detail')
+      .text()
+      .should.be.empty()
     })
 
-    it('adds a detail last child', () => {
-      shallow(<Label detail={faker.hacker.noun()}><br /></Label>)
-        .children()
-        .last()
-        .should.match('.detail')
-    })
-
-    it('adds the value as the detail text', () => {
-      const detail = faker.hacker.noun()
-      shallow(<Label detail={detail} />)
-        .find('.detail')
-        .should.contain.text(detail)
-    })
-  })
+    it('adds the value as children', () => {
+      const text = faker.hacker.phrase()
 
-  describe('icon', () => {
-    it('adds a i as first child', () => {
-      shallow(<Label icon={faker.hacker.noun()}><br /></Label>)
-        .childAt(0)
-        .should.match('i')
+      shallow(<Label content={text} />)
+        .should.contain.text(text)
     })
   })
 
-  describe('image', () => {
-    it('has no img when prop is not defined', () => {
+  describe('detail', () => {
```

Comment:
Let's replace this with `common.implementsShorthandProp()` for the LabelDetail.  Reference other tests for examples, or see #530.